rules/REQUEST-922-MULTIPART-ATTACK.conf

```diff
@@ -40,8 +40,9 @@ SecRule &MULTIPART_PART_HEADERS:_charset_ "!@eq 0" \
     tag:'paranoia-level/1',\
     ver:'OWASP_CRS/4.0.0-rc2',\
     severity:'CRITICAL',\
+    setvar:'tx.922100_charset=|%{ARGS._charset_}|',\
```

Comment:
Not sure about it, but would be possible to write a test that fails before the fix and works after?

Reply:
@M4tteoP About exactly what are you not sure?

Reply:
I think he's asking for a test case, which I think is a good idea too. Here it is:

```yaml
  - test_title: 922100-3
    desc: "Negative test: don't allow substrings of valid charsets when using _charset_"
    stages:
      - stage:
          input:
            dest_addr: 127.0.0.1
            headers:
              Host: "localhost"
              User-Agent: "OWASP CRS test agent"
              Content-Type: multipart/form-data; boundary=boundary
              Accept: "*/*"
            method: POST
            port: 80
            uri: "/post"
            version: "HTTP/1.1"
            data: |
              --boundary
              Content-disposition: form-data; name="_charset_"

              utf-
              --boundary
              Content-disposition: form-data; name="negative"
              Content-Type: text/plain

              utf-8 is valid but utf- should not be.
              --boundary--
          output:
            no_log_contains: id "922100"

```

Reply:
Yes, I was asking for a test case. Thanks!